### DISCUSSION TITLE
Checking correlation between numeric and categoric column.

### DISCUSSION THREAD

**Discussion by MF-Faqih** [teamalgoritma/community#728] [Q&A] [answered]
Saya ingin melakukan cek korelasi antar prediktor, prediktor yang saya miliki ada yang bertipe numerik dan ada yang bertipa kategorik. Bagaimana caranya untuk melakukan test korelasi antar prediktor ini? Saya mencoba mengguankan MANOVA untuk melakuakn test korelasi, namun karna terdapat beberapa kolom yang tidak mengikuti distribusi normal, maka hasilnya cukup aneh. Berikut adalah hasil yang didapatkan dari uji MANOVA.

![ask](https://github.com/teamalgoritma/community/assets/105436454/08e714ed-a084-4af3-a43d-f33d43a6d6a3)

apakah ada cara lainnya untuk melakukan check korelasi antar prediktor ini? atau ada hal lain yang bisa saya lakukan untuk memperbaiki test MANOVA ini?

**Comment by ranydwi** [comment]
Dear Pak @MF-Faqih 
Untuk mencari nilai korelasi antara variabel kategorik dan numerik dapat memanfaatkan korelasi biserial. 
Di R, korelasi biserial dapat dicari menggunakan bantuan package `ltm`

```
library(ltm)
set.seed(123)
x <- rnorm(100)
y <- factor(sample(c("A", "B"), 100, replace = TRUE))
biserial.cor(x, y)
```
Contohnya adalah code berikut di atas
Semoga membantu

**Reply by MF-Faqih** [reply]
apakah ada cara lain  jika kolom kategorik dan numerik yang saya miliki berjumlah lebih dari satu? Atau harus dilakuakn cek korelasi satu persatu?

**Reply by ranydwi** [reply, marked as answer]
Dear Pak Faqih, 
Terima kasih untuk kesediaannya menunggu.

Untuk MANOVA sendiri bukan digunakan untuk melihat atau mengecek korelasi. Hal ini karena tujuan dari MANOVA adalah untuk melihat ada atau tidaknya perbedaan rata-rata antara beberapa variabel sekaligus. Dalam statistik, pengujian korelasi untuk kategorik bisa memanfaatkan biserial itu. Namun kekurangannya di R, dia hanya bisa one-by-one variabel. Sehingga belum ada cara untuk bisa melakukan pengecekan korelasi biserial dalam sekaligus.
Semoga membantu
